repos/spack_repo/builtin/packages/curl/package.py

```diff
@@ -86,27 +92,33 @@ class Curl(NMakePackage, AutotoolsPackage):
     elif sys.platform == "win32":
         default_tls = "sspi"
 
-    # TODO: add dependencies for other possible TLS backends
-    variant(
-        "tls",
-        default=default_tls,
-        description="TLS backend",
-        values=(
-            # 'amissl',
-            # 'bearssl',
-            "gnutls",
-            conditional("mbedtls", when="@7.46:"),
-            # 'mesalink',
-            conditional("nss", when="@:7.81"),
-            "openssl",
-            # 'rustls',
-            # 'schannel',
-            "secure_transport",
-            # 'wolfssl',
-            conditional("sspi", when="platform=windows"),
-        ),
-        multi=True,
+    tls_desc = "TLS backend"
+    tls_vals = (
+        # 'amissl',
+        # 'bearssl',
+        "gnutls",
+        conditional("mbedtls", when="@7.46:"),
+        # 'mesalink',
+        conditional("nss", when="@:7.81"),
+        "openssl",
+        # 'rustls',
+        # 'schannel',
+        # secure_transport support was removed in curl 8.15.0
+        conditional("secure_transport", when="platform=darwin @:8.14"),
+        # 'wolfssl',
+        conditional("sspi", when="platform=windows"),
     )
+
+    # TODO: add dependencies for other possible TLS backends
+
+    variant("tls", default="openssl", description=tls_desc, values=tls_vals, multi=True)
+    with when("platform=windows"):
+        variant("tls", default="sspi", description=tls_desc, values=tls_vals, multi=True)
+    with when("platform=darwin @:8.14"):
+        variant(
+            "tls", default="secure_transport", description=tls_desc, values=tls_vals, multi=True
+        )
```

Comment:
If we set defaults here, in terms of conditional variants, we can remove the `default_tls` definition above